functions/New-DbaDbSequence.ps1

```diff
@@ -0,0 +1,190 @@
+        Shows what would happen if the command were to run. No actions are actually performed.
+
+    .PARAMETER Confirm
+        Prompts you for confirmation before executing any changing operations within the command.
+
+    .PARAMETER EnableException
+        By default, when something goes wrong we try to catch it, interpret it and give you a friendly warning message.
+        This avoids overwhelming you with "sea of red" exceptions, but is inconvenient because it basically disables advanced scripting.
+        Using this switch turns this "nice by default" feature off and enables you to catch exceptions with your own try/catch.
+
+    .NOTES
+        Tags: Data, Database, Migration, Sequence, Table
+        Author: Adam Lancaster https://github.com/lancasteradam
+
+        dbatools PowerShell module (https://dbatools.io)
+        Copyright: (c) 2021 by dbatools, licensed under MIT
+        License: MIT https://opensource.org/licenses/MIT
+
+    .LINK
+        https://dbatools.io/New-DbaDbSequence
+
+    .EXAMPLE
+        PS C:\> New-DbaDbSequence -SqlInstance sqldev01 -Database TestDB -Name TestSequence -StartWith 10000 -IncrementBy 10
+
+        Creates a new sequence TestSequence in the TestDB database on the sqldev01 instance. The sequence will start with 10000 and increment by 10.
+
+    .EXAMPLE
+        PS C:\> New-DbaDbSequence -SqlInstance sqldev01 -Database TestDB -Name TestSequence -Cycle
+
+        Creates a new sequence TestSequence in the TestDB database on the sqldev01 instance. The sequence will cycle the numbers.
+
+    .EXAMPLE
+        PS C:\> Get-DbaDatabase -SqlInstance sqldev01 -Database TestDB | New-DbaDbSequence -Name TestSequence -Schema TestSchema -IntegerType bigint
+
+        Using a pipeline this command creates a new bigint sequence named TestSchema.TestSequence in the TestDB database on the sqldev01 instance.
+    #>
+    [CmdletBinding(SupportsShouldProcess, ConfirmImpact = 'Low')]
+    param (
+        [DbaInstanceParameter[]]$SqlInstance,
+        [PSCredential]$SqlCredential,
+        [string[]]$Database,
+        [Parameter(Mandatory)]
+        [string]$Name,
+        [string]$Schema = 'dbo',
+        [string]$IntegerType = 'bigint',
+        [long]$StartWith = 1,
+        [long]$IncrementBy = 1,
+        [long]$MinValue,
+        [long]$MaxValue,
+        [switch]$Cycle,
+        [int32]$CacheSize,
+        [parameter(ValueFromPipeline)]
+        [Microsoft.SqlServer.Management.Smo.Database[]]$InputObject,
+        [switch]$EnableException
+    )
+    process {
+
+        if ((Test-Bound -ParameterName SqlInstance) -and (Test-Bound -Not -ParameterName Database)) {
+            Stop-Function -Message "Database is required when SqlInstance is specified"
+            return
+        }
+
+        foreach ($instance in $SqlInstance) {
+            $InputObject += Get-DbaDatabase -SqlInstance $instance -SqlCredential $SqlCredential -Database $Database
+        }
+
+        foreach ($db in $InputObject) {
```

Comment:
This command would be best to work per instance instead of per database. Unless you found this to be just as performant?

Consider if user provides 12 instances and each of those instances has 15 databases, how does this command handle that processing? That is a lot of object to cache and then iterate over versus caching 15 objects on each instead instead of 180 databases in one go.

Reply:
I'm glad you mentioned this. Yes, in this case it does perform well. Creating a sequence in 300 databases took about 25 seconds on my dev machine and there was no abnormal memory usage.

I see what you mean about processing the batch per instance and I agree that is better. However, this pattern of doing the "$InputObject +=" seems to be the common approach for supporting both the pipeline input and the normal command invocation with parameters.